### ISSUE TITLE
Set as Startpage

### ISSUE THREAD

**Issue by mrprobot** [MatMoul/quickdial-webext#21]
An option to set Quick Dial as startpage would be nice. I managed to set it via Firefox 56, because there you can see the path to the extension in the adressbar, but in Firefox 57 I actually had no idea how to set it.

**Comment by scienmind** [comment]
Also, as mentioned [here](https://github.com/MatMoul/quickdial-webext/issues/15#issuecomment-337007333), it would be very useful to be able to access Quick Dial using a install-independent URL (i.e. not `moz-extension://IDIDID-IDID...IDID-IDIDID/dial`), so that can be synced to other devices, which will have a different ID for the extension.

**Comment by MatMoul** [comment]
Currently, you can open your settings page and add a new tab, then go back to settings page and click to current page.
But I'll try to read the doc if I can provide a button to set a page as start page.

**Comment by scienmind** [comment]
> Currently, you can open your settings page and add a new tab, then go back to settings page and click to current page.

That just sets home page to `moz-extension://IDIDID-IDID...IDID-IDIDID/dial`, as I mentioned. Both on FF56 and FF58 (nightly) - and that's the issue: `IDIDID-IDID...IDID-IDIDID` part is device-dependent, and syncing the Home Page like that breaks functionality.

I tried finding a work around, but no success so far. 

What needed is a way to bind a universal address to the extension. Like `about:something` or `moz-extension://quick_dial` (synced) that'd redirect to `moz-extension://IDIDID-IDID...IDID-IDIDID/dial` (per device) for example

**Comment by MatMoul** [comment]
I enjoy your solution but I'm not sure that it's possible with webext restrictions...

**Comment by evgeny-test** [comment]
This problem is very urgent for me.
Maybe it will be useful for you.
In Fast Dial there was a special link "chrome://fastdial/content/fastdial.html".
Thank you!

**Comment by MatMoul** [comment]
@evgeny-test In my knowledge, webext don't provide mechanism to do this.
I'm thinking hard how I can sync data with other device and this problematic is linked...

**Comment by scienmind** [comment]
FYI: my current workaround:  I just exclude "homepage" from sync using one of the FF's flags, and set it manually on each device. This way I can still have bookmarks synced without dysfunctional homepage.

I hope a more proper solution will be found for that.

**Comment by evgeny-test** [comment]
@scienmind What option do you use to exclude only the home page?

**Comment by scienmind** [comment]
@evgeny-test : 
about:config -> set `services.sync.prefs.sync.browser.startup.homepage` to `false`
[(source)](https://support.mozilla.org/en-US/questions/919674)

**Comment by MatMoul** [comment]
I think I found a pretty clean solution but that is only compatible with ff57+
I just need time to implement it and to be sure that don't break esr version...

**Comment by mazesy** [comment]
I'm looking forwards -- anxiously! -- to the option of setting QuickDial as FF's homepage.

At this time I've set QuickDial (its moz-extension wording) as FF's homepage and it works ok except that it takes like two seconds to have the startpage operational when it is otherwise quasi instantaneous with FF's default homepage, mainly because FF transits the page via its newtab routine (I tested with a 10 line htm file fed with only background color & image and it was delayed as well...).

Quick Dial is gorgeous, need to say. Makes a perfect New Tab page. But I use only New Tab/Quick Dial so having it popup on start would be wunderbar :)

Thanks Matmoul for this beautiful code/extension.
